x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/edit_field/update_field_provider.tsx

```diff
@@ -77,24 +77,40 @@ export const UpdateFieldProvider = ({ children }: Props) => {
     };
 
     if (field.source.type !== previousField.source.type) {
-      const aliases = getAllDescendantAliases(field, fields)
-        .map(id => byId[id].path)
-        .sort();
-      const hasAliases = Boolean(aliases.length);
+      const aliasesOnField = aliases[field.id];
+      const aliasesOnFieldAndDescendants = getAllDescendantAliases(field, fields);
+      const doSomeAliasPointsOnCurrentField = aliasesOnField && Boolean(aliasesOnField.length);
```

Comment:
Perhaps something like `isReferencedByAlias` would read more easily? I know it's a tricky one though! :)

Reply:
I like that, I will change it. Thanks for the suggestion! 👍